odl/solvers/advanced/forward_backward.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Implementation of the forward-backward primal-dual splitting algorithm for
+optimization.
+"""
+
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+from odl.operator import Operator
+
+
+__all__ = ('forward_backward_pd',)
+
+
+def forward_backward_pd(x, prox_f, prox_cc_g, L, grad_h, tau, sigma, niter,
+                        callback=None, **kwargs):
+    """ The forward-backward primal-dual splitting algorithm.
+
+    The algorithm minimizes the sum of several convex functionals composed with
+    linear operators,
+
+        ``min_x f(x) + sum_i g_i(L_i x) + h(x)``,
+
+    where ``f``, ``g_i`` are convex functionals, ``L_i`` are linear
+    `Operator`'s, and ``h`` is a convex and differentiable functional.
```

Comment:
I wouldn't make a link in all places. Here it seems to look better to just use the word "operators".